src/main/java/com/couchbase/lite/replicator/ChangeTracker.java

```diff
@@ -295,214 +288,161 @@ protected void runLoop() {
-                        final Credentials credentials = new UsernamePasswordCredentials(username, password);
-
-                        if (httpClient instanceof DefaultHttpClient) {
-                            DefaultHttpClient dhc = (DefaultHttpClient) httpClient;
-                            HttpRequestInterceptor preemptiveAuth = new HttpRequestInterceptor() {
-                                @Override
-                                public void process(HttpRequest request, HttpContext context) throws HttpException, IOException {
-                                    AuthState authState = (AuthState) context.getAttribute(ClientContext.TARGET_AUTH_STATE);
-                                    if (authState.getAuthScheme() == null) {
-                                        authState.setAuthScheme(new BasicScheme());
-                                        authState.setCredentials(credentials);
-                                    }
-                                }
-                            };
-                            dhc.addRequestInterceptor(preemptiveAuth, 0);
-                        }
-                    } else {
-                        Log.w(Log.TAG_CHANGE_TRACKER, "RemoteRequest Unable to parse user info, not setting credentials");
-                    }
-                }
+            // Perform BASIC Authentication if needed
+            builder = RequestUtils.preemptivelySetAuthCredentials(builder, url, authenticator);
+            request = builder.build();
 
+            try {
+                String maskedRemoteWithoutCredentials = getChangesFeedURL().toString();
+                maskedRemoteWithoutCredentials = maskedRemoteWithoutCredentials.replaceAll("://.*:.*@", "://---:---@");
+                Log.v(Log.TAG_CHANGE_TRACKER, "%s: Making request to %s", this, maskedRemoteWithoutCredentials);
+                call = httpClient.newCall(request);
+                Response response = call.execute();
                 try {
-                    String maskedRemoteWithoutCredentials = getChangesFeedURL().toString();
-                    maskedRemoteWithoutCredentials = maskedRemoteWithoutCredentials.replaceAll("://.*:.*@", "://---:---@");
-
-                    Log.v(Log.TAG_CHANGE_TRACKER, "%s: Making request to %s", this, maskedRemoteWithoutCredentials);
-                    HttpResponse response = httpClient.execute(request);
-
                     // In case response status is Error, ChangeTracker stops here
-                    if(isResponseFailed(response)) {
-                        if(retryIfFailedPost(response.getStatusLine()))
+                    if (isResponseFailed(response)) {
+                        if (retryIfFailedPost(response)) {
+                            RequestUtils.closeResponseBody(response);
                             continue;
+                        }
+                        RequestUtils.closeResponseBody(response);
```

Comment:
RequestUtils.closeResponseBody(response) in line 326 and 323 can be moved up to before line 322.